### DISCUSSION TITLE
FIX NAME INTERFACE PPPoE WITH CHARACTER <name>

### DISCUSSION THREAD

**Discussion by kevindoni** [hotspotbilling/phpnuxbill#206] [General]
Line 144 - 163 in mikrotik.php

![13](https://github.com/hotspotbilling/phpnuxbill/assets/22675011/2e8967c7-8f82-47a5-bf72-79125ce60883)

    $interfaceData = [];
    foreach ($traffic as $interface) {
        $name = $interface->getProperty('name');
        // Skip interfaces with missing names
        if (empty($name)) {
            continue;
        }

        $txBytes = intval($interface->getProperty('tx-byte'));
        $rxBytes = intval($interface->getProperty('rx-byte'));

        $interfaceData[] = [
            'name' => $name,
            'status' => $interface->getProperty('running') === 'true' ? '
            <small class="label bg-green">up</small>' : '
            <small class="label bg-red">down</small>',
            'tx' => mikrotik_formatBytes($txBytes),
            'rx' => mikrotik_formatBytes($rxBytes)
        ];
    }

REPLAY WITH

    $interfaceData = [];
    foreach ($traffic as $interface) {
        $name = $interface->getProperty('name');
    
        $txBytes = intval($interface->getProperty('tx-byte'));
        $rxBytes = intval($interface->getProperty('rx-byte'));
    
        // Konversi karakter khusus ke dalam entitas HTML
        $name = htmlspecialchars($name);
    
        $interfaceData[] = [
            'name' => $name,
            'status' => $interface->getProperty('running') === 'true' ? '
            <small class="label bg-green">up</small>' : '
            <small class="label bg-red">down</small>',
            'tx' => mikrotik_formatBytes($txBytes),
            'rx' => mikrotik_formatBytes($rxBytes)
        ];
    }

DONE

![12](https://github.com/hotspotbilling/phpnuxbill/assets/22675011/11e8fe01-af73-4c30-9033-fb494cba540b)

**Comment by ibnux** [comment]
@Focuslinkstech this is your plugin

**Reply by Focuslinkstech** [reply]
i have fix it since uncle @ibnux but i refused to release it, because someone promise me that when i fix it he will donate and he refused to do so, thats why i refused to release it

![image](https://github.com/hotspotbilling/phpnuxbill/assets/45756999/66f494de-5ed2-49da-9bf1-299e310432b8)

i have even fix the upload, download and total usages that its not available on the previous version

![image](https://github.com/hotspotbilling/phpnuxbill/assets/45756999/f1d3db60-d5c6-4ab6-bb65-0b6ac4e13152)


![image](https://github.com/hotspotbilling/phpnuxbill/assets/45756999/8b53079d-4166-4405-b3b7-5b5374eefe2e)



i  will release it on behalf of you @ibnux

**Comment by alucard2303** [comment]
puedes poner botones para activar o desactivar los usuarios pppoe por favor

**Reply by Focuslinkstech** [reply]
Alright i will consider that in the future updates.
Thanks
